src/Microsoft.OData.Edm/Csdl/CsdlReader.Json.cs

```diff
@@ -0,0 +1,312 @@
+            CsdlJsonReaderOptions options = new CsdlJsonReaderOptions
+            {
+                JsonReaderFactory = referencedModelReaderFunc
+            };
+
+            return TryParse(ref reader, options, out model, out errors);
+        }
+
+        /// <summary>
+        /// Tries parsing the given CSDL-JSON artifact for an IEdmModel.
+        /// </summary>
+        /// <param name="reader">The given JSON reader containing the CSDL artifact.</param>
+        /// <param name="references">Models to be referenced by the created model.</param>
+        /// <param name="model">The model generated by parsing</param>
+        /// <param name="errors">Errors reported while parsing.</param>
+        /// <returns>Success of the parse operation.</returns>
+        public static bool TryParse(ref Utf8JsonReader reader, IEnumerable<IEdmModel> references, out IEdmModel model, out IEnumerable<EdmError> errors)
+        {
+            CsdlJsonReaderOptions options = new CsdlJsonReaderOptions
+            {
+                ReferencedModels = references
+            };
+
+            return TryParse(ref reader, options, out model, out errors);
+        }
+
+        /// <summary>
+        /// Tries parsing the given CSDL-JSON artifact for an IEdmModel.
+        /// </summary>
+        /// <param name="reader">The given JSON reader containing the CSDL artifact.</param>
+        /// <param name="references">Models to be referenced by the created model.</param>
+        /// <param name="includeDefaultVocabularies">A value indicating enable/disable the built-in vocabulary supporting.</param>
+        /// <param name="model">The model generated by parsing</param>
+        /// <param name="errors">Errors reported while parsing.</param>
+        /// <returns>Success of the parse operation.</returns>
+        public static bool TryParse(ref Utf8JsonReader reader, IEnumerable<IEdmModel> references, bool includeDefaultVocabularies, out IEdmModel model, out IEnumerable<EdmError> errors)
+        {
+            CsdlJsonReaderOptions options = new CsdlJsonReaderOptions
+            {
+                ReferencedModels = references,
+                IncludeDefaultVocabularies = includeDefaultVocabularies
+            };
+
+            return TryParse(ref reader, options, out model, out errors);
+        }
+
+        /// <summary>
+        /// Tries parsing the given CSDL-JSON artifact for an IEdmModel.
+        /// </summary>
+        /// <param name="reader">The given JSON reader containing the CSDL artifact.</param>
+        /// <param name="settings">CSDL-JSON reader settings for current parser.</param>
+        /// <param name="model">The model generated by parsing</param>
+        /// <param name="errors">Errors reported while parsing.</param>
+        /// <returns>Success of the parse operation.</returns>
+        public static bool TryParse(ref Utf8JsonReader reader, CsdlJsonReaderOptions options, out IEdmModel model, out IEnumerable<EdmError> errors)

```

Comment:
Would this not better be the last parameter  to make this more consistent with 
`TryParse(ref Utf8JsonReader reader, out IEdmModel model, out IEnumerable<EdmError> errors)`

Reply:
Emm, i have to put all non-out parameter before the out parameter. So, I'd like to keep it unchanged.

---
In reply to: [425980401](https://github.com/OData/odata.net/pull/1775#discussion_r425980401) [](ancestors = 425980401)